examples/fluids/qfunctions/newtonian_state.h

```diff
@@ -132,6 +132,23 @@ CEED_QFUNCTION_HELPER StateConservative StateConservativeFromPrimitive_fwd(Newto
   return dU;
 }
 
+CEED_QFUNCTION_HELPER StateConservative StateConservativeAXPBY(CeedScalar a, StateConservative X, CeedScalar b, StateConservative Y) {
+  StateConservative R;
+  R.density = a * X.density + b * Y.density;
+  for (int i = 0; i < 3; i++) R.momentum[i] = a * X.momentum[i] + b * Y.momentum[i];
+  R.E_total = a * X.E_total + b * Y.E_total;
+  return R;
+}
+
+CEED_QFUNCTION_HELPER StateConservative StateConservativeAXPBYPCZ(CeedScalar a, StateConservative X, CeedScalar b, StateConservative Y, CeedScalar c,
```

Comment:
Is there anyway to make the state structs indexable, so this kind of function wouldn't be `StateConservative` specific? So we'd be able to do something like
```c
for (int i=0; i<5; i++) {
  R[i] = a*X[i] + b*Y[i];
}
```
and thus have this be for any state struct? I've been looking into it and there are some weird hacks with `union` operators. Main issue we run into is padding for structs, but that can be overcome with attributes. But I'm not sure if we really want to tie ourselves to a key part of functioning code being formed by compiler-dependent features.

Reply:
Not really in C. You can get away with a union or cast on most compilers/opt levels, but it's undefined behavior and the attributes are not really portable. C++ and Rust can do it soundly.